.github/workflows/main.yml

```diff
@@ -0,0 +1,51 @@
+name: Main
+
+on:
+  pull_request: {}
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-22.04
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version-file: go.mod
+
+    - name: Go pkgs check
+      run: make vet
+
+    - name: Build meme and memtierd
+      run: |
+        make bin/memtierd
+        make bin/meme
+
+    - name: Unit test
+      run: make race=1 test
+
+    - name: Fuzz test
+      run: make race=1 fuzztest
+
+    - name: Install dependencies
```

Comment:
Nit: slightly more descriptive name would be better: install dependencies for what? ("static checks"?)